source/common/event/dispatcher_impl.cc

```diff
@@ -314,6 +349,14 @@ void DispatcherImpl::runPostCallbacks() {
     // callback executes.
     callbacks.pop_front();
   }
+
+  while (!to_be_delete.empty()) {
```

Comment:
These deletions should be done in a different callback, not runPostCallbacks

Reply:
done